tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java

```diff
@@ -43,25 +39,26 @@ protected SetType(TiColumnInfo.InternalTypeHolder holder) {
   /** {@inheritDoc} */
   @Override
   protected Object decodeNotNull(int flag, CodecDataInput cdi) {
-    throw new UnsupportedTypeException("Set type not supported");
+    if (flag != Codec.UVARINT_FLAG) throw new TypeException("Invalid IntegerType flag: " + flag);
+    return this.getElems().get((int) IntegerCodec.readUVarLong(cdi) - 2);
```

Comment:
Why minus 2 here? Would you please add comments here?

Reply:
same question.

Reply:
Parsing Set logic is different with paring enum logic. I just updated my implementation.